Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py

```diff
@@ -151,3 +152,42 @@ def retrieve_last_run_test_index_from_output(test_spec_list: list[EditorTestBase
         else:
             index += 1
     return index
+
+def save_failed_asset_joblogs(workspace: AbstractWorkspace) -> None:
+    """
+    Checks all asset logs in the JobLogs directory to see if the asset has any warnings or errors. If so, the asset is
+    saved via ArtifactManager.
+
+    :param workspace: The AbstractWorkspace to access the JobLogs path
+    :return: None
+    """
+    # Tuples are in the form [this directory, [child directories], [files in this directory]]
+    for walk_tuple in os.walk(workspace.paths.ap_job_logs()):
+        for log_file in walk_tuple[2]:
+            full_log_path = os.path.join(walk_tuple[0], log_file)
+            # Only save asset logs that contain errors or warnings
+            if _check_log_errors_warnings(full_log_path):
+                try:
+                    workspace.artifact_manager.save_artifact(full_log_path)
+                except Exception as e:  # Purposefully broad
+                    logger.warning(e)
```

Comment:
This should log a human-readable warning that the artifact (at the given path) is being skipped, due to the error. Else the logging has little context.

I dislike the pattern of broad catch-and-discard, but this cleanup script is in fact an appropriate place for it.

Reply:
improved logging